From efe4b0d09e4bfca69e210ab0f6c0487098a667db Mon Sep 17 00:00:00 2001
From: Hardik Panchal <hardikx.panchal@intel.com>
Date: Tue, 15 Nov 2022 11:22:38 +0000
Subject: [PATCH] Add retimer image type

This commit adds the new enum for retimer firmware updates in PFR image
types. Also add newly added type for verification of image.

Tested:
 - Uploaded image from redfish and image verification is success and
   passed image to next layers i.e fwupd.sh

Signed-off-by: hmudun1x <hemanthx.kumar.mudunuri@intel.com>
Signed-off-by: AppaRao Puli <apparao.puli@intel.com>
Signed-off-by: Hardik Panchal <hardikx.panchal@intel.com>
Upstream-Status: Pending
---
 pfr_image_manager.cpp | 6 ++++--
 pfr_image_manager.hpp | 3 ++-
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/pfr_image_manager.cpp b/pfr_image_manager.cpp
index 73c8371..dffe5f8 100644
--- a/pfr_image_manager.cpp
+++ b/pfr_image_manager.cpp
@@ -205,8 +205,10 @@ int Manager::verifyImage(const std::filesystem::path imgPath,
             {
                 purposeString = verPurpose + "Host";
             }
-            else if (imgType == pfrCPLDUpdateCap || imgType == pfrAFMUpdateCap ||
-                     imgType == pfrCmpstCPLDUpdCap)
+            else if (imgType == pfrCPLDUpdateCap ||
+                     imgType == pfrAFMUpdateCap ||
+                     imgType == pfrCmpstCPLDUpdCap ||
+                     imgType == pfrCmpstRetimerUpdCap)
             {
                 purposeString = verPurpose + "Other";
             }
diff --git a/pfr_image_manager.hpp b/pfr_image_manager.hpp
index 99fb9e3..046cace 100644
--- a/pfr_image_manager.hpp
+++ b/pfr_image_manager.hpp
@@ -24,7 +24,8 @@ enum pfrImgPCType
     pfrBMCUpdateCap = 0x04,
     seamlessPCHUpdCap = 0x05,
     pfrAFMUpdateCap = 0x06,
-    pfrCmpstCPLDUpdCap = 0x07
+    pfrCmpstCPLDUpdCap = 0x07,
+    pfrCmpstRetimerUpdCap = 0x09
 };
 
 /* PFR image block 0 - As defined in HAS */
-- 
2.25.1

